module_auto_update/models/module.py

```diff
@@ -39,36 +36,30 @@ def _compute_checksum_dir(self):
                 excluded_extensions=exclude
             )
 
-    @api.depends('state')
-    def _compute_checksum_installed(self):
-        stored_checksums = {}
-        for r in self.read(['checksum_installed']):
-            stored_checksums[r['id']] = r['checksum_installed']
-
-        for r in self:
-            if r.state == 'installed':
+    def _store_checksum_installed(self, vals):
+        if (vals.get('state') == 'installed' and
+                not self.env.context.get('retain_checksum_installed')):
+            for r in self:
                 r.checksum_installed = r.checksum_dir
-            else:
-                r.checksum_installed = stored_checksums[r.id]
 
     @api.multi
     def button_uninstall_cancel(self):
-        return self.retain_checksum_installed(
-            super(Module, self).button_uninstall_cancel
-        )
+        return super(
+            Module,
+            self.with_context(retain_checksum_installed=True)
+        ).button_uninstall_cancel()
 
     @api.multi
     def button_upgrade_cancel(self):
-        return self.retain_checksum_installed(
-            super(Module, self).button_upgrade_cancel
-        )
+        return super(
+            Module,
+            self.with_context(retain_checksum_installed=True)
+        ).button_upgrade_cancel()
 
-    def retain_checksum_installed(self, func):
-        installed_checksums = self.mapped('checksum_installed')
-        res = func()
-        for index, r in enumerate(self):
-            r.checksum_installed = installed_checksums[index]
-        return res
+    @api.model
+    def create(self, vals):
+        self._store_checksum_installed(vals)
+        return super(Module, self).create(vals)
```

Comment:
First, create the record, and then call to `result._store_checksum_installed(vals)`, or it will have no effect.